src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakPickerHiRes.cpp

```diff
@@ -347,10 +364,21 @@ namespace OpenMS
         {
           continue;
         }
+
         CubicSpline2d peak_spline (peak_raw_data);
 
         // calculate maximum by evaluating the spline's 1st derivative
         // (bisection method)
+        // Since we included situations where left or right neighbors could not exist
+        // we could create a 'phantom' neighbor peak of equal distance from central mz to existing neighbor mz
+        // but how to handle ion mobility array? spline_bisection expects "peak_spline" to have the same mz range
+        // Introducing a 'phantom' peak extending past 'peak_spline' range will throw an error.
+        // We have to add a proper peak to peak_spline and have to decide what ion mobility value to assign to it.
+
+        // for now, simply replace mz value of missing neighbor with central mz peak.
+        left_neighbor_mz = (no_left_neighbor) ? central_peak_mz: left_neighbor_mz;
```

Comment:
I think this could be fine.
What is repoted if report_FWHM is enabled?

Reply:
https://github.com/OpenMS/OpenMS/blob/16a93d924ac776a730a05ab9f68fd815f5e1599e/src/openms/include/OpenMS/MATH/MISC/SplineBisection.h#L73
with your solution now, it will start in the middle and probably move up to the top where the derivative gets close to zero and yield the correct result.
If you add a phantom peak it might error because it evaluates the derivative exactly at or slightly outside the peak range. 

I think how you do it is fine with our implementation.

Reply:
> I think this could be fine.
> What is repoted if report_FWHM is enabled?

Here is a screenshot with the FWHM reported in ppm. 

<img width="1507" alt="Screen Shot 2023-04-06 at 12 59 02 PM" src="https://user-images.githubusercontent.com/63276677/230446056-17337582-e772-46cd-ba36-ceae75d3f9cc.png">